File: examples/android/src/main/java/com/dropbox/core/examples/android/ListFolderTask.java
package com.dropbox.core.examples.android;

import android.os.AsyncTask;

import com.dropbox.core.DbxException;
import com.dropbox.core.v2.DbxFiles;

/**
 * Async task to list items in a folder
 */
class ListFolderTask extends AsyncTask<String, Void, DbxFiles.ListFolderResult> {

    private final DbxFiles mDbxFilesClient;
    private Exception mException;
    private Callback mCallback;

    public interface Callback {
        void onDataLoaded(DbxFiles.ListFolderResult result);

        void onError(Exception e);
    }

    public ListFolderTask(DbxFiles DbxFilesClient, Callback callback) {
        mDbxFilesClient = DbxFilesClient;
        mCallback = callback;
    }

    @Override
    protected void onPostExecute(DbxFiles.ListFolderResult result) {
        super.onPostExecute(result);

        if (mException != null) {
            mCallback.onError(mException);
        } else {
            mCallback.onDataLoaded(result);
        }
    }

    @Override
    protected DbxFiles.ListFolderResult doInBackground(String... params) {
        try {
            return mDbxFilesClient.listFolder(params[0]);
        } catch (DbxException e) {
            mException = e;
            e.printStackTrace();
        }

        return null;
    }
}

File: examples/android/src/main/java/com/dropbox/core/examples/android/ListFolderTask.kt
package com.dropbox.core.examples.android

import com.dropbox.core.DbxException
import com.dropbox.core.v2.DbxClientV2
import com.dropbox.core.v2.files.ListFolderResult
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.withContext

sealed interface ListFolderApiResult {
    class Success(val result: ListFolderResult) : ListFolderApiResult
    class Error(val e: Exception) : ListFolderApiResult
}

/**
 * Async task to list items in a folder
 */
internal class ListFolderTask(
    private val mDbxClient: DbxClientV2,
) {
    suspend fun execute(path: String): ListFolderApiResult = withContext(Dispatchers.IO) {
        try {
            ListFolderApiResult.Success(mDbxClient.files().listFolder(path))
        } catch (e: DbxException) {
            ListFolderApiResult.Error(e)
        }
    }
}